### PR TITLE
Dot_product comments fix

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR adds some information in the documentation for dot products.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/polynomial_ring_zq/dot_product.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    /// let poly_1 = PolynomialRingZq::from_str("4  1 0 2 1 / 5  1 2 3 4 5 mod 11").unwrap();
+    /// let poly_2 = PolynomialRingZq::from_str("1  9 / 5  1 2 3 4 5 mod 11").unwrap();
+    ///
+    /// let dot_prod = poly_1.dot_product(&poly_2).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    ///     [`MathError::MismatchingModulus`] if the moduli mismatch.
+    pub fn dot_product(&self, other: &Self) -> Result<Zq, MathError> {
```
Comment (jnsiemer) [inline src/integer_mod_q/polynomial_ring_zq/dot_product.rs:38]: I couldn't find a definition for the dot_product for polynomials by a quick Google search. Can you share a link to a definition and add a reference to it in the doc comment? Thanks

**Review by Marvin-Beckmann — APPROVED** [review]
